doc/scripts/autogen-docs

```diff
@@ -10,19 +10,32 @@ ROOTDIR="$(cd "$( dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)/../.."
 BUILDDIR="${ROOTDIR}/build"
 SPICYDOC="${BUILDDIR}/bin/spicy-doc"
 SPICYDTR="${ROOTDIR}/doc/scripts/spicy-doc-to-rst"
-AUTOGEN="${ROOTDIR}/doc/autogen"
+AUTOGEN_FINAL="${ROOTDIR}/doc/autogen"
+AUTOGEN_STAGE="${ROOTDIR}/doc/autogen.$$.tmp"
```

Comment:
Can we use `mktemp` here? This would remove most errors around the directory existing, and also give us a robust way to put the directory into the system temp dir.

    mktemp -d -t spicy-autogen-docs